specification/dsdl/serialization.tex

```diff
@@ -614,6 +622,11 @@ \subsubsection{Delimited types}\label{sec:dsdl_serialization_composite_non_seale
 and the implicit zero extension (section \ref{sec:dsdl_serialization_implicit_zero_extension})
 rules apply.
 
+The length encoded in a delimiter header cannot exceed the number of bytes remaining between the delimiter header
+and the end of the serialized representation of the outer object.
```

Comment:
Is this property transitive between all outer objects? i.e. should this read:
```suggestion
The length encoded in a delimiter header cannot exceed the number of bytes remaining between the delimiter header
and the end of the serialized representation of the top-level object.
```

Reply:
This formulation implies that the length of the serialized representation of the next outer object in the hierarchy is irrelevant, which is incorrect. Suppose A (top-level object) nests B (outer object) which nests C. If the delimiter header of C specifies sizeof(C) bytes but the serialized representation of B ends in less than sizeof(C) bytes, the serialized representation of the entire structure (A,B, and C) is invalid. In your formulation, this would not be the case because it applies only to the top-level object A ignoring the direct container of C.

Reply:
yeah. you're right.